internal/metadata/icon.go

```diff
@@ -0,0 +1,41 @@
+package metadata
+
+import (
+	"bytes"
+	"image/png"
+	"strconv"
+
+	"github.com/nfnt/resize"
+
+	"fyne.io/fyne/v2"
+)
+
+func ScaleIcon(data fyne.Resource, size int) fyne.Resource {
+	img, err := png.Decode(bytes.NewReader(data.Content()))
+	if err != nil {
+		fyne.LogError("Failed to decode app icon", err)
+		return data
+	}
+
+	if img.Bounds().Dx() <= size {
```

Comment:
How do we handle non-square icons? We are only checking the X dimension here, it's possible the Y dimension could be larger, unless we explicitly don't support non-square?

Reply:
Only square icons are supported